### PR TITLE
feat(pipeline): edgepipeline_register providers add ShouldDispatchToEdge method

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
edgepipeline_register providers add ShouldDispatchToEdge method

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      edgepipeline_register providers add ShouldDispatchToEdge method        |
| 🇨🇳 中文    |      edgepipeline_register providers 添加 ShouldDispatchToEdge 方法        |

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]
why move from `edgepipeline` provider to `edgepipeline_register`?
In general, other providers should use `edgepipeline` directly, instead of `edgepipeline_register`.

**Comment by kakj-go** [comment]
> why move from `edgepipeline` provider to `edgepipeline_register`? In general, other providers should use `edgepipeline` directly, instead of `edgepipeline_register`.

edgepipeline have pipelinesvc import , Circular dependency occurs when other providers refer to this provider

**Review by chengjoey — COMMENTED** [review]
`edgepipeline_register` and `edgepipeline` providers may should merge into one later,
otherwise looks stranger.
`edgepipeline_register` should only provider register function

**Comment by sfwn** [comment]
> > why move from `edgepipeline` provider to `edgepipeline_register`? In general, other providers should use `edgepipeline` directly, instead of `edgepipeline_register`.
> 
> edgepipeline have pipelinesvc import , Circular dependency occurs when other providers refer to this provider

You should resolve this circle dependency issue, instead of move specific method.

File: modules/pipeline/providers/edgepipeline_register/provider.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 type Config struct {
 	IsEdge                     bool          `env:"DICE_IS_EDGE" default:"false"`
 	ClusterName                string        `env:"DICE_CLUSTER_NAME"`
+	AllowedSources             []string      `env:"EDGE_ALLOW_SOURCES"`
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/provider.go:32]: EDGE_ALLOWED_SOURCES

and remove this config in original `edgepipeline`.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**